core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -264,6 +265,8 @@ class ReplicaManager(val config: KafkaConfig,
     leaderPartitionsIterator.count(_.hasLateTransaction(currentTimeMs))
   }
 
+  def producerIdCount: Int = onlinePartitionsIterator.map(_.producerIdCount).sum
```

Comment:
Topic level metrics are a bit tricky because each broker may not host all the partitions of a give topic. However, I suppose it could give a general idea of tenant level usage. However, one concern is cardinality and the way that we would aggregate and store these topic level metrics. KIP-847 also mentions the rejected alternative of partition level metrics, but another concern there was cardinality.

Reply:
Thanks, got it. Exposing a partial aggregate per topic per broker may be misleading indeed, and we can use the new RPC `DescribeProducers` from KIP 847 can be used to get the additional level of observability.